File: priv/templates/companies/bench-softdev/fixtures/repo-go/theme_controller.go
package benchfixture

import "sync"

// Theme state. Bench task bugs-go-2 extends this to support
// "dark" and to round-trip the value.
//
// Intentionally skeletal — the bench task's job is to extend it.

var (
	themeMu      sync.RWMutex
	currentTheme = "light"
)

// SetTheme sets the current theme. Only "light" is accepted
// today; bugs-go-2 should add "dark" plus an error on invalid
// input.
func SetTheme(theme string) {
	themeMu.Lock()
	defer themeMu.Unlock()
	if theme == "light" {
		currentTheme = "light"
	}
}

// CurrentTheme returns the currently-set theme.
func CurrentTheme() string {
	themeMu.RLock()
	defer themeMu.RUnlock()
	return currentTheme
}

File: priv/templates/companies/bench-softdev/fixtures/repo/lib/theme_controller.ex
defmodule BenchFixture.ThemeController do
  @moduledoc """
  In-process theme state. Bench task bugs-2 extends this to support
  :dark and to round-trip through `current_theme/0`.

  NOTE: intentionally skeletal — the bench task's job is to extend
  this module.
  """

  use Agent

  def start_link(_) do
    Agent.start_link(fn -> :light end, name: __MODULE__)
  end

  def set_theme(:light), do: Agent.update(__MODULE__, fn _ -> :light end)

  def current_theme, do: Agent.get(__MODULE__, & &1)
end
